checker/src/main/java/org/checkerframework/checker/nullness/NullnessNoInitVisitor.java

```diff
@@ -767,6 +768,23 @@ public Void visitSwitch(SwitchTree tree, Void p) {
         if (!TreeUtils.hasNullCaseLabel(tree)) {
             checkForNullability(tree.getExpression(), SWITCHING_NULLABLE);
         }
+        if (checker.getLintOption(
+                NullnessChecker.LINT_REDUNDANTNULLCOMPARISON,
+                NullnessChecker.LINT_DEFAULT_REDUNDANTNULLCOMPARISON)) {
+            ExpressionTree expression = tree.getExpression();
+            List<? extends CaseTree> cases = tree.getCases();
+            AnnotatedTypeMirror switchType = atypeFactory.getAnnotatedType(expression);
+            for (CaseTree caseTree : cases) {
+                @SuppressWarnings("deprecation")
+                ExpressionTree caseExpression = caseTree.getExpression();
```

Comment:
I used `getExpression()` because jkd 8, 11 do not have the methods `getExpressions()`. And the `getExpression()`  has been deprecated in jdk 17 and above.

Reply:
Isn't it enough to use the `TreeUtils.hasNullCaseLabel(tree)` that's already computed for the previous `if`?
Just put that result into a local variable and reuse it here.

For helper methods for these version-specific getters, see `TreeUtilsAfterJava11`.

Reply:
> Isn't it enough to use the `TreeUtils.hasNullCaseLabel(tree)` that's already computed for the previous `if`? Just put that result into a local variable and reuse it here.
> 
> For helper methods for these version-specific getters, see `TreeUtilsAfterJava11`.

I think we still need logic here to locate which case should we report the error to.